source/Subsystem/AbilitySystem.cpp

```diff
@@ -0,0 +1,84 @@
+#include "AbilitySystem.h"
+#include "EntityComponentSystem.h"
+
+bool AbilitySystem::StartAbilityOnEntity(Ability anAbility, const UniqueID aTargetEntity)
+{
+	Entity* affectedEntity = myEntityComponentSystem->GetEntity(aTargetEntity);
+	if (!affectedEntity || !anAbility.myEffect)
+		return false;
+
+	bool affectTagsMatch = false;
+	
+	for (auto tag : anAbility.myAffectTags)
+	{
+		if (tag == AbilityTag::Any || HasTag(tag, aTargetEntity))
+		{
+			affectTagsMatch = true;
+			break;
+		}
+	}
+
+	if (!affectTagsMatch)
+		return false;
+	
+	if (anAbility.myAffectOverridesIgnore)
+	{
+		anAbility.myEffect();
+		return true;
+	}
+
+	for (auto tag : anAbility.myIgnoreTags)
```

Comment:
If I remember, you wanted to refrain from using auto for most things, so consider using the proper type here.

Reply:
Can also be const.

Reply:
Fixed.